pyvista/plotting/charts.py

```diff
@@ -0,0 +1,3375 @@
+    >>> chart.x_axis.tick_locations = x
+    >>> chart.x_axis.tick_labels = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", 
+    ...                             "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
+    >>> chart.x_label = "Month"
+    >>> chart.y_axis.tick_labels = "2e"
+    >>> chart.y_label = "# incidents"
+    >>> chart.show()
+
+    """
+
+    PLOT_TYPES = {
+        "scatter": ScatterPlot2D,
+        "line": LinePlot2D,
+        "area": AreaPlot,
+        "bar": BarPlot,
+        "stack": StackPlot
+    }
+
+    def __init__(self, size=(1, 1), loc=(0, 0), x_label="x", y_label="y", grid=True):
+        """Initialize the chart."""
+        super().__init__()
+        self._plots = {plot_type: [] for plot_type in self.PLOT_TYPES.keys()}
+        self.SetAutoSize(False)  # We manually set the appropriate size
+        self.size = size
+        self.loc = loc
+        # self.SetAxis(_vtk.vtkAxis.BOTTOM, self._x_axis)  # Disabled for now and replaced by a wrapper object, as for
+        # self.SetAxis(_vtk.vtkAxis.LEFT, self._y_axis)  # some reason vtkChartXY.SetAxis(...) causes a crash at the end
+        self._x_axis = Axis(_wrap=self.GetAxis(_vtk.vtkAxis.BOTTOM))  # of the script's execution (nonzero exit code)
+        self._y_axis = Axis(_wrap=self.GetAxis(_vtk.vtkAxis.LEFT))
+        self.x_label = x_label
+        self.y_label = y_label
+        self.grid = grid
+        self.legend_visible = True
+
+    def _render_event(self, *args, **kwargs):
+        self.RecalculateBounds()
+        super()._render_event(*args, **kwargs)
+
+    def _add_plot(self, plot_type, *args, **kwargs):
+        plot = self.PLOT_TYPES[plot_type](*args, **kwargs)
+        self.AddPlot(plot)
+        self._plots[plot_type].append(plot)
+        return plot
+
+    @classmethod
+    def parse_format(cls, fmt):
+        """Parse a format string and separate it into a marker style, line style and color.
+
+        Parameters
+        ----------
+        fmt : str
+            Format string to parse. A format string consists of any combination of a valid marker
+            style, a valid line style and parsable color. The specific order does not matter. See
+            :attr:`pyvista.ScatterPlot2D.MARKER_STYLES` for a list of valid marker styles,
+            :attr:`pyvista.Pen.LINE_STYLES` for a list of valid line styles and
+            :func:`pyvista.parse_color` for an overview of parsable colors.
+
+        Returns
+        -------
+        marker_style : str
+            Extracted marker style (empty string if no marker style was present in the format string).
+
+        line_style : str
+            Extracted line style (empty string if no line style was present in the format string).
+
+        color : str
+            Extracted color string (defaults to ``"b"`` if no color was present in the format string).
+
+        Examples
+        --------
+        >>> m, l, c = pyvista.Chart2D.parse_format("x--b")
```

Comment:
And there's a doctest error coming from a missing import here:
```suggestion
        >>> import pyvista
        >>> m, l, c = pyvista.Chart2D.parse_format("x--b")
```
It would also be nice to say something about this example.  What does it exemplify?